src/test/java/net/jodah/typetools/TypeResolverTest.java

```diff
@@ -86,12 +87,53 @@ public void shouldResolveArgumentForList() {
     assertEquals(TypeResolver.resolveRawArgument(List.class, SomeList.class), Integer.class);
   }
 
+  public void shouldResolveTypeForList() {
+    Type resolvedType = TypeResolver.substituteGenerics(List.class, SomeList.class);
+    assert resolvedType instanceof ParameterizedType;
+    assertEquals(((ParameterizedType) resolvedType).getActualTypeArguments()[0], Integer.class);
+  }
+
   public void shouldResolveArgumentsForBazFromFoo() {
     Class<?>[] typeArguments = TypeResolver.resolveRawArguments(Baz.class, Foo.class);
     assert typeArguments[0] == HashSet.class;
     assert typeArguments[1] == ArrayList.class;
   }
 
+  public void shouldResolveParameterizedTypeForBazFromFoo() {
+    Type type = TypeResolver.substituteGenerics(Baz.class, Foo.class);
+
+    // Now we walk the type hierarchy:
+    assert type instanceof ParameterizedType;
+    Type[] typeArguments = ((ParameterizedType) type).getActualTypeArguments();
+
+    assert typeArguments[0] instanceof ParameterizedType;
+    ParameterizedType firstTypeArgument = (ParameterizedType) typeArguments[0];
+    assert firstTypeArgument.getRawType() == HashSet.class;
+    assert firstTypeArgument.getActualTypeArguments()[0] == Object.class;
```

Comment:
This test would be better if the type argument was something other than Object.

Reply:
Well, in this case there's no explicit upper bound for the wildcard type, but I still wanted to re-use the existing fixture. I would like to keep this test, and making it "better" would require a different fixture.

Maybe we can add another one transports better what you want. So you want one case where a wildcard type is resolved to something below `Object`? I added this in aa744c9. If that's not what you mean, please give clearer instructions.

Reply:
> So you want one case where a wildcard type is resolved to something below Object?

Yea, I thought that would be more clear.

> I would like to keep this test, and making it "better" would require a different fixture.


Good point.